daemon/cluster/executor/backend.go

```diff
@@ -76,7 +76,7 @@ type VolumeBackend interface {
 
 // ImageBackend is used by an executor to perform image operations
 type ImageBackend interface {
-	PullImage(ctx context.Context, image, tag string, platform *ocispec.Platform, metaHeaders map[string][]string, authConfig *registry.AuthConfig, outStream io.Writer) error
+	PullImage(ctx context.Context, ref reference.Named, platform *ocispec.Platform, metaHeaders map[string][]string, authConfig *registry.AuthConfig, outStream io.Writer) error
```

Comment:
No, we digested can still be passed as `reference.Named` so pull by digest (eg. `docker pull busybox@sha256:1411f4a8c78f5fadafa8f733e71f6ff01dfd637263ae090d68511a6e152451e3` will work).

However, this wouldn't accept `docker pull sha256:1411f4a8c78f5fadafa8f733e71f6ff01dfd637263ae090d68511a6e152451e3`, which isn't a valid reference anyway?

Reply:
Was actually wondering about that last one the other day (because everything is content-addressable) if it's technically possible to get a "digest" from a registry (if they were to allow it). But maybe that's not a thing as it would make it difficult (impossible) to check if you should have access to that digest